src/main.rs

```diff
@@ -2519,6 +2519,27 @@ fn run_steps(ib: &mut ImageBuilder) -> Result<()> {
                     &a.publisher,
                 ])?;
             }
+            "pkg_set_p5p_publisher" => {
+                #[derive(Deserialize)]
+                struct PkgSetP5PPublisherArgs {
+                    archive: String,
+                    publisher: String,
+                }
+
+                let a: PkgSetP5PPublisherArgs = step.args()?;
+                let mp = ib.root()?;
+
+                pkg(log, &["-R", &mp.to_str().unwrap(),
+                    "set-publisher", "-p", &a.archive, "--search-first",
+                ])?;
+                pkg(log, &["-R", &mp.to_str().unwrap(), "set-publisher",
+                    "--no-refresh",
+                    "--non-sticky",
+                    &a.publisher,
+                ])?;
+                pkg(log, &["-R", &mp.to_str().unwrap(), "refresh", "--full"])?;
+                pkg(log, &["-R", &mp.to_str().unwrap(), "update"])?;
```

Comment:
I've been talking to @jclulow about similar things.  and feel the need to add that if you just did `refresh --full`, you probably want to pass `--no-refresh` to everything else, to avoid doing checks again and again.  Here, especially, that seems easy.  It might be less easy for every pkg operation (but it would be good!)